repos/system_upgrade/common/actors/peseventsscanner/libraries/pes_event_parsing.py

```diff
@@ -0,0 +1,252 @@
+    DEPRECATED = 2
+    REPLACED = 3
+    SPLIT = 4
+    MERGED = 5
+    MOVED = 6
+    RENAMED = 7
+
+
+def get_pes_events(pes_json_directory, pes_json_filename):
+    """
+    Get all the events from the source JSON file exported from PES.
+
+    :return: List of Event tuples, where each event contains event type and input/output pkgs
+    """
+    try:
+        all_events = parse_pes_events(fetch.read_or_fetch(pes_json_filename, directory=pes_json_directory,
+                                                          allow_empty=True))
+        arch = api.current_actor().configuration.architecture
+        events_matching_arch = [e for e in all_events if not e.architectures or arch in e.architectures]
+        return events_matching_arch
+    except (ValueError, KeyError):
+        title = 'Missing/Invalid PES data file ({}/{})'.format(pes_json_directory, pes_json_filename)
+        summary = 'Read documentation at: https://access.redhat.com/articles/3664871 for more information ' \
+            'about how to retrieve the files'
+        reporting.create_report([
+            reporting.Title(title),
+            reporting.Summary(summary),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Tags([reporting.Tags.SANITY]),
+            reporting.Flags([reporting.Flags.INHIBITOR]),
+            reporting.RelatedResource('file', os.path.join(pes_json_directory, pes_json_filename))
+        ])
+        raise StopActorExecution()
+
+
+def generate_event_for_ms_mapping_entry(from_ms_to_ms_entry, event):
+    from_ms, to_ms = from_ms_to_ms_entry
+    return Event(
+        event.id,
+        event.action,
+        {Package(p.name, p.repository, from_ms) for p in event.in_pkgs if from_ms in p.modulestream},
+        {Package(p.name, p.repository, to_ms) for p in event.out_pkgs if to_ms in p.modulestream},
+        event.from_release,
+        event.to_release,
+        event.architectures
+    )
+
+
+def event_by_modulestream_mapping(mapping, event):
+    """
+    Generate events with matching event.ID according to the the given modulestream mapping.
+
+    Modulestream allows for compact symbolic representation of multiple events having the same input/output
+    packages differing only in the modules they are in. For example consider event renaming package X with
+    module streams [MSX0, MSX1], to a new package with the name Y and possible module streams [MSY0, MSY1].
+    The attached modulestream map {MSX0 -> MSY0, MSX1 -> MSY1} would mean that the renaming event represents
+    two events --- one renaming (X, ms=MSX0) to (Y, ms=MSY0), and the second one renaming (X, ms=MSX1)
+    to (Y, ms=MSY1).
+
+
+    :param mapping: Dictionary mapping input package modulestreams to output modulestreams.
+    :param event: PES event that should be expanded.
+    :returns: A list of generated PES events.
+    """
+    if not mapping:
+        # In case there is no mapping, assume all is going to non modular
+        mapping = {}
```

Comment:
I know it's late to ask for this, but please start using a Dict[str, List[str]] instead of Dict[str, str] dictionary - I am fine with a follow up to add it, just please we need to do this, due to changes to PES

Reply:
I have changed the code to collect all `to_modulestreams` per `from_modulestream`,  @vinzenz please check that my understanding was correct, as I had to refresh what were the changes in PES about.